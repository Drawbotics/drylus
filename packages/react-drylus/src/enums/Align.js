import Enum from '@drawbotics/enums';


const Align = new Enum(
  'LEFT',
  'CENTER',
  'RIGHT',
);


export default Align;
