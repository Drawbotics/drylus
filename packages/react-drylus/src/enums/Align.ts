export enum Align {
  LEFT = 'LEFT',
  CENTER = 'CENTER',
  RIGHT = 'RIGHT',
  JUSTIFY = 'JUSTIFY',
}
