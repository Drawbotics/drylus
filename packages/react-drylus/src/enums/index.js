export Category from './Category';
export Tier from './Tier';
export Size from './Size';
export Position from './Position';
export Align from './Align';
export Color from './Color';
export Shade from './Shade';
