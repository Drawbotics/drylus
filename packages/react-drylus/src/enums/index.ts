export * from './Category';
export * from './Tier';
export * from './Size';
export * from './Position';
export * from './Align';
export * from './Color';
export * from './Shade';
