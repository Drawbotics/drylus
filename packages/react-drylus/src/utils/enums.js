import camelCase from 'lodash/camelCase';

import { Category, Color, Shade, Tier } from '../enums';

export function getEnumAsClass(enumVal) {
  return camelCase(enumVal?.description?.toLowerCase());
}

export function colorEnumToCategory(enumVal) {
  switch (enumVal) {
    case Color.RED:
      return Category.DANGER;
    case Color.GREEN:
      return Category.SUCCESS;
    case Color.ORANGE:
      return Category.WARNING;
    case Color.BLUE:
      return Category.INFO;
  }
  return enumVal;
}

export function categoryEnumToColor(enumVal) {
  switch (enumVal) {
    case Category.DANGER:
      return Color.RED;
    case Category.SUCCESS:
      return Color.GREEN;
    case Category.WARNING:
      return Color.ORANGE;
    case Category.INFO:
      return Color.BLUE;
  }
  return enumVal;
}

export function shadeEnumToTier(enumVal) {
  switch (enumVal) {
    case Shade.DARK:
      return Tier.PRIMARY;
    case Shade.MEDIUM:
      return Tier.SECONDARY;
    case Shade.LIGHT:
      return Tier.TERTIARY;
  }
  return enumVal;
}

export function getIconForCategory(category) {
  switch (category) {
    case Category.DANGER:
      return 'alert-circle';
    case Category.SUCCESS:
      return 'check-circle';
    case Category.WARNING:
      return 'alert-triangle';
    default:
      return 'info';
  }
}
