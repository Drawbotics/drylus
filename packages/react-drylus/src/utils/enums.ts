import camelCase from 'lodash/camelCase';

import { Category, Color, Position, Shade, Size, Tier } from '../enums';

type Enum = Category | Color | Shade | Tier | Size | Position;

export function getEnumAsClass<T extends Record<string, string>>(enumVal?: Enum): keyof T {
  return camelCase(enumVal?.toLowerCase());
}

export function colorEnumToCategory(enumVal: Color): Category | undefined {
  switch (enumVal) {
    case Color.RED:
      return Category.DANGER;
    case Color.GREEN:
      return Category.SUCCESS;
    case Color.ORANGE:
      return Category.WARNING;
    case Color.BLUE:
      return Category.INFO;
  }
}

export function categoryEnumToColor(enumVal: Category): Color | undefined {
  switch (enumVal) {
    case Category.DANGER:
      return Color.RED;
    case Category.SUCCESS:
      return Color.GREEN;
    case Category.WARNING:
      return Color.ORANGE;
    case Category.INFO:
      return Color.BLUE;
  }
}

export function shadeEnumToTier(enumVal: Shade): Tier | undefined {
  switch (enumVal) {
    case Shade.DARK:
      return Tier.PRIMARY;
    case Shade.MEDIUM:
      return Tier.SECONDARY;
    case Shade.LIGHT:
      return Tier.TERTIARY;
  }
}

export function getIconForCategory(category: Category): string {
  switch (category) {
    case Category.DANGER:
      return 'alert-circle';
    case Category.SUCCESS:
      return 'check-circle';
    case Category.WARNING:
      return 'alert-triangle';
    default:
      return 'info';
  }
}
