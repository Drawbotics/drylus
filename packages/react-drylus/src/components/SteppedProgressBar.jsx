import sv from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import PropTypes from 'prop-types';
import React from 'react';

import { Category, Size } from '../enums';
import { getEnumAsClass } from '../utils';
import { useResponsiveProps } from '../utils/hooks';

const styles = {
  root: css`
    display: flex;
    height: ${sv.marginExtraSmall};
    width: 100%;
    position: relative;
    border-radius: ${sv.marginExtraSmall};
    overflow: hidden;
    background: ${sv.neutralLight};

    [data-element='bar'] {
      border-radius: ${sv.marginExtraSmall};
    }
  `,
  large: css`
    height: ${sv.marginSmall};
    border-radius: ${sv.marginSmall};

    [data-element='bar'] {
      border-radius: ${sv.marginSmall};
    }
  `,
  extraLarge: css`
    height: 90px;
  `,
  small: css`
    height: calc(${sv.marginExtraSmall} / 2);
    border-radius: calc(${sv.marginExtraSmall} / 2);

    [data-element='bar'] {
      border-radius: calc(${sv.marginExtraSmall} / 2);
    }
  `,
  step: css`
    flex: 1;
    position: relative;
  `,
  bar: css`
    position: absolute;
    z-index: 2;
    top: 0;
    left: 0;
    height: 100%;
    width: 0;
    background: ${sv.neutralDark};
    transition: ${sv.transitionShort};
    overflow: hidden;
  `,
};

const SteppedProgressBar = ({ responsive, ...rest }) => {
  const { percentage, category, size, style, steps, activeStep } = useResponsiveProps(
    rest,
    responsive,
  );

  const indeterminate = percentage == null;

  return (
    <div style={style} className={cx(styles.root, { [styles[getEnumAsClass(size)]]: size })}>
      {[...Array(steps).keys()].map((id) => (
        <div className={styles.step} key={id}>
          <div
            className={cx(styles.stepChild, {
              [styles.active]: id === activeStep,
              [styles.indeterminate]: indeterminate,
              [styles[getEnumAsClass(category)]]: category,
            })}
            style={{ width: `${percentage * 100}%` }}
            data-element="bar"
          />
        </div>
      ))}
    </div>
  );
};

SteppedProgressBar.propTypes = {
  /** Determines how many steps there are in the bar */
  steps: PropTypes.number.required,

  /** The index of the currenctly active step */
  activeStep: PropTypes.number.required,

  /** If specified the currently active bar has a precise width, should be between 0-1 */
  progress: PropTypes.number,

  category: PropTypes.oneOf([
    Category.BRAND,
    Category.DANGER,
    Category.SUCCESS,
    Category.INFO,
    Category.WARNING,
  ]),

  size: PropTypes.oneOf([Size.SMALL, Size.DEFAULT, Size.LARGE]),

  /** Used for style overrides */
  style: PropTypes.object,

  /** Reponsive prop overrides */
  responsive: PropTypes.shape({
    XS: PropTypes.object,
    S: PropTypes.object,
    M: PropTypes.object,
    L: PropTypes.object,
    XL: PropTypes.object,
    HUGE: PropTypes.object,
  }),
};

SteppedProgressBar.defaultProps = {
  size: Size.DEFAULT,
};

export default SteppedProgressBar;
