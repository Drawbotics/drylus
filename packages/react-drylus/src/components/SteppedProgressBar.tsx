import sv from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import React, { Fragment } from 'react';

import { Category, Color, Size } from '../enums';
import { Responsive, Style } from '../types';
import { Deprecated, categoryEnumToColor, getEnumAsClass, run, useResponsiveProps } from '../utils';
import { placeholderStyles as shimmerStyles } from './LoadingPlaceholder';

const styles = {
  root: css`
    display: flex;
    height: ${sv.marginExtraSmall};
    width: 100%;
    position: relative;
    border-radius: ${sv.marginExtraSmall};
    overflow: hidden;
    background: ${sv.neutralLight};

    [data-element='step']:first-child {
      [data-element='bar'] {
        border-top-left-radius: ${sv.marginExtraSmall};
        border-bottom-left-radius: ${sv.marginExtraSmall};
      }
    }

    [data-element='step']:last-child {
      [data-element='bar'] {
        border-top-right-radius: ${sv.marginExtraSmall};
        border-bottom-right-radius: ${sv.marginExtraSmall};
      }
    }
  `,
  large: css`
    height: ${sv.marginSmall};
    border-radius: ${sv.marginSmall};

    [data-element='step']:first-child {
      [data-element='bar'] {
        border-top-left-radius: ${sv.marginSmall};
        border-bottom-left-radius: ${sv.marginSmall};
      }
    }

    [data-element='step']:last-child {
      [data-element='bar'] {
        border-top-right-radius: ${sv.marginSmall};
        border-bottom-right-radius: ${sv.marginSmall};
      }
    }
  `,
  extraLarge: css`
    height: 90px;
  `,
  small: css`
    height: calc(${sv.marginExtraSmall} / 2);
    border-radius: calc(${sv.marginExtraSmall} / 2);

    [data-element='step']:first-child {
      [data-element='bar'] {
        border-top-left-radius: calc(${sv.marginExtraSmall} / 2);
        border-bottom-left-radius: calc(${sv.marginExtraSmall} / 2);
      }
    }

    [data-element='step']:last-child {
      [data-element='bar'] {
        border-top-right-radius: calc(${sv.marginExtraSmall} / 2);
        border-bottom-right-radius: calc(${sv.marginExtraSmall} / 2);
      }
    }
  `,
  step: css`
    flex: 1;
    position: relative;
  `,
  bar: css`
    position: absolute;
    z-index: 2;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    background: ${sv.neutralDark};
    transition: ${sv.transitionShort};
    overflow: hidden;

    &::after {
      background: ${sv.neutralDark};
      border-radius: 0 !important;
      background-image: linear-gradient(
        to right,
        ${sv.neutralDark} 8%,
        ${sv.neutral} 18%,
        ${sv.neutralDark} 33%
      ) !important;
    }
  `,
  separator: css`
    height: 100%;
    width: 1px;
    background: ${sv.white};
  `,
  brand: css`
    background: ${sv.brand};

    &::after {
      background: ${sv.brand};
      background-image: linear-gradient(
        to right,
        ${sv.brand} 8%,
        ${sv.brandLight} 18%,
        ${sv.brand} 33%
      ) !important;
    }
  `,
  red: css`
    background: ${sv.red};

    &::after {
      background: ${sv.red};
      background-image: linear-gradient(
        to right,
        ${sv.red} 8%,
        ${sv.redLight} 18%,
        ${sv.red} 33%
      ) !important;
    }
  `,
  blue: css`
    background: ${sv.blue};

    &::after {
      background: ${sv.blue};
      background-image: linear-gradient(
        to right,
        ${sv.blue} 8%,
        ${sv.blueLight} 18%,
        ${sv.blue} 33%
      ) !important;
    }
  `,
  orange: css`
    background: ${sv.orange};

    &::after {
      background: ${sv.orange};
      background-image: linear-gradient(
        to right,
        ${sv.orange} 8%,
        ${sv.orangeLight} 18%,
        ${sv.orange} 33%
      ) !important;
    }
  `,
  green: css`
    background: ${sv.green};

    &::after {
      background: ${sv.green};
      background-image: linear-gradient(
        to right,
        ${sv.green} 8%,
        ${sv.greenLight} 18%,
        ${sv.green} 33%
      ) !important;
    }
  `,
  active: css`
    border-radius: 0 !important;
  `,
};

export interface SteppedProgressBarProps {
  /** Determines how many steps there are in the bar */
  steps: number;

  /** The index of the currenctly active step */
  activeStep: number;

  /** If specified the currently active bar has a precise width, should be between 0-1 */
  percentage?: number;

  /**
   * @deprecated use color instead
   * @kind Category
   */
  category?: Category.BRAND | Category.SUCCESS | Category.INFO | Category.WARNING | Category.DANGER;

  /** @kind Color */
  color?: Color.BRAND | Color.RED | Color.BLUE | Color.GREEN | Color.ORANGE;

  /**
   * @default Size.DEFAULT
   * @kind Size
   */
  size?: Size.SMALL | Size.DEFAULT | Size.LARGE;

  /** Used for style overrides */
  style?: Style;

  /** Reponsive prop overrides */
  responsive?: Responsive<this>;
}

export const SteppedProgressBar = ({ responsive, ...rest }: SteppedProgressBarProps) => {
  const {
    percentage = 0,
    category,
    size = Size.DEFAULT,
    style,
    steps,
    activeStep,
    color: _color,
  } = useResponsiveProps<SteppedProgressBarProps>(rest, responsive);

  const indeterminate = percentage == null;
  const color = category ? categoryEnumToColor(category) : _color;

  return (
    <div
      style={style}
      className={cx(styles.root, { [styles[getEnumAsClass<typeof styles>(size)]]: size != null })}>
      {[...Array(steps).keys()].map((id, i) => (
        <Fragment key={id}>
          <div data-element="step" className={styles.step}>
            <div
              className={cx(styles.bar, {
                [styles.active]: id == activeStep && percentage !== 1,
                [styles[getEnumAsClass<typeof styles>(color)]]: color != null,
                [shimmerStyles.shimmer]: id == activeStep && indeterminate,
              })}
              style={{
                width: run(() => {
                  if (id < activeStep || (id == activeStep && indeterminate)) {
                    return '100%';
                  } else if (id == activeStep) {
                    return `${percentage * 100}%`;
                  } else {
                    return '0';
                  }
                }),
              }}
              data-element="bar"
            />
          </div>
          {run(() => {
            if (i !== steps - 1) {
              return <div className={styles.separator} />;
            }
          })}
        </Fragment>
      ))}
    </div>
  );
};

SteppedProgressBar.propTypes = {
  category: Deprecated,
};
