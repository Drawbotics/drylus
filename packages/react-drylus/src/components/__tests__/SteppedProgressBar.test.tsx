import React from 'react';
import { create } from 'react-test-renderer';

import { Color, Size } from '../../enums';
import { SteppedProgressBar } from '../SteppedProgressBar';

describe('SteppedProgressBar', () => {
  describe('matches snapshot when', () => {
    it('has 4 steps', () => {
      const tree = create(<SteppedProgressBar steps={4} activeStep={0} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a color', () => {
      const tree = create(
        <SteppedProgressBar steps={4} activeStep={0} color={Color.BRAND} />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('is small', () => {
      const tree = create(
        <SteppedProgressBar steps={4} activeStep={0} size={Size.SMALL} />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a percentage', () => {
      const tree = create(
        <SteppedProgressBar steps={4} activeStep={0} percentage={0.4} />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
