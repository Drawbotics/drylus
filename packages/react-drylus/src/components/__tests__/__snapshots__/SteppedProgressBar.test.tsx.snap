// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SteppedProgressBar matches snapshot when has 4 steps 1`] = `
@keyframes animation-0 {
  0% {
    background-position: -1200px 0;
  }

  100% {
    background-position: 1200px 0;
  }
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 8px;
  width: 100%;
  position: relative;
  border-radius: 8px;
  overflow: hidden;
  background: #eaeff4;
}

.emotion-11 [data-element='step']:first-child [data-element='bar'] {
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
}

.emotion-11 [data-element='step']:last-child [data-element='bar'] {
  border-top-right-radius: 8px;
  border-bottom-right-radius: 8px;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

.emotion-3 {
  position: absolute;
  z-index: 2;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background: #879ba6;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
}

.emotion-3::after {
  background: #879ba6;
  border-radius: 0 !important;
  background-image: linear-gradient( to right, #879ba6 8%, #d3dce6 18%, #879ba6 33% ) !important;
}

.emotion-2 {
  height: 100%;
  width: 1px;
  background: #fff;
}

.emotion-0 {
  position: absolute;
  z-index: 2;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background: #879ba6;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
  border-radius: 0 !important;
  position: relative;
  pointer-events: none;
}

.emotion-0::after {
  background: #879ba6;
  border-radius: 0 !important;
  background-image: linear-gradient( to right, #879ba6 8%, #d3dce6 18%, #879ba6 33% ) !important;
}

.emotion-0::after {
  content: ' ';
  position: absolute;
  z-index: 9;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  background: #d3dce6;
  border-radius: 4px;
  overflow: hidden;
  background-image: linear-gradient( to right, #d3dce6 8%, #eaeff4 18%, #d3dce6 33% );
  background-size: 1200px 100%;
  -webkit-animation: animation-0 2s forwards infinite linear;
  animation: animation-0 2s forwards infinite linear;
}

<div
  className="emotion-11 undefined"
>
  <div
    className="emotion-1"
    data-element="step"
  >
    <div
      className="emotion-0"
      data-element="bar"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="emotion-2"
  />
  <div
    className="emotion-1"
    data-element="step"
  >
    <div
      className="emotion-3"
      data-element="bar"
      style={
        Object {
          "width": "0",
        }
      }
    />
  </div>
  <div
    className="emotion-2"
  />
  <div
    className="emotion-1"
    data-element="step"
  >
    <div
      className="emotion-3"
      data-element="bar"
      style={
        Object {
          "width": "0",
        }
      }
    />
  </div>
  <div
    className="emotion-2"
  />
  <div
    className="emotion-1"
    data-element="step"
  >
    <div
      className="emotion-3"
      data-element="bar"
      style={
        Object {
          "width": "0",
        }
      }
    />
  </div>
</div>
`;

exports[`SteppedProgressBar matches snapshot when has a color 1`] = `
@keyframes animation-0 {
  0% {
    background-position: -1200px 0;
  }

  100% {
    background-position: 1200px 0;
  }
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 8px;
  width: 100%;
  position: relative;
  border-radius: 8px;
  overflow: hidden;
  background: #eaeff4;
}

.emotion-11 [data-element='step']:first-child [data-element='bar'] {
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
}

.emotion-11 [data-element='step']:last-child [data-element='bar'] {
  border-top-right-radius: 8px;
  border-bottom-right-radius: 8px;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

.emotion-2 {
  height: 100%;
  width: 1px;
  background: #fff;
}

.emotion-0 {
  position: absolute;
  z-index: 2;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background: #879ba6;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
  border-radius: 0 !important;
  background: #3DB5D0;
  position: relative;
  pointer-events: none;
}

.emotion-0::after {
  background: #879ba6;
  border-radius: 0 !important;
  background-image: linear-gradient( to right, #879ba6 8%, #d3dce6 18%, #879ba6 33% ) !important;
}

.emotion-0::after {
  background: #3DB5D0;
  background-image: linear-gradient( to right, #3DB5D0 8%, #bee6ef 18%, #3DB5D0 33% ) !important;
}

.emotion-0::after {
  content: ' ';
  position: absolute;
  z-index: 9;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  background: #d3dce6;
  border-radius: 4px;
  overflow: hidden;
  background-image: linear-gradient( to right, #d3dce6 8%, #eaeff4 18%, #d3dce6 33% );
  background-size: 1200px 100%;
  -webkit-animation: animation-0 2s forwards infinite linear;
  animation: animation-0 2s forwards infinite linear;
}

.emotion-3 {
  position: absolute;
  z-index: 2;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background: #879ba6;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
  background: #3DB5D0;
}

.emotion-3::after {
  background: #879ba6;
  border-radius: 0 !important;
  background-image: linear-gradient( to right, #879ba6 8%, #d3dce6 18%, #879ba6 33% ) !important;
}

.emotion-3::after {
  background: #3DB5D0;
  background-image: linear-gradient( to right, #3DB5D0 8%, #bee6ef 18%, #3DB5D0 33% ) !important;
}

<div
  className="emotion-11 undefined"
>
  <div
    className="emotion-1"
    data-element="step"
  >
    <div
      className="emotion-0"
      data-element="bar"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="emotion-2"
  />
  <div
    className="emotion-1"
    data-element="step"
  >
    <div
      className="emotion-3"
      data-element="bar"
      style={
        Object {
          "width": "0",
        }
      }
    />
  </div>
  <div
    className="emotion-2"
  />
  <div
    className="emotion-1"
    data-element="step"
  >
    <div
      className="emotion-3"
      data-element="bar"
      style={
        Object {
          "width": "0",
        }
      }
    />
  </div>
  <div
    className="emotion-2"
  />
  <div
    className="emotion-1"
    data-element="step"
  >
    <div
      className="emotion-3"
      data-element="bar"
      style={
        Object {
          "width": "0",
        }
      }
    />
  </div>
</div>
`;

exports[`SteppedProgressBar matches snapshot when has a percentage 1`] = `
.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 8px;
  width: 100%;
  position: relative;
  border-radius: 8px;
  overflow: hidden;
  background: #eaeff4;
}

.emotion-11 [data-element='step']:first-child [data-element='bar'] {
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
}

.emotion-11 [data-element='step']:last-child [data-element='bar'] {
  border-top-right-radius: 8px;
  border-bottom-right-radius: 8px;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

.emotion-3 {
  position: absolute;
  z-index: 2;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background: #879ba6;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
}

.emotion-3::after {
  background: #879ba6;
  border-radius: 0 !important;
  background-image: linear-gradient( to right, #879ba6 8%, #d3dce6 18%, #879ba6 33% ) !important;
}

.emotion-2 {
  height: 100%;
  width: 1px;
  background: #fff;
}

.emotion-0 {
  position: absolute;
  z-index: 2;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background: #879ba6;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
  border-radius: 0 !important;
}

.emotion-0::after {
  background: #879ba6;
  border-radius: 0 !important;
  background-image: linear-gradient( to right, #879ba6 8%, #d3dce6 18%, #879ba6 33% ) !important;
}

<div
  className="emotion-11 undefined"
>
  <div
    className="emotion-1"
    data-element="step"
  >
    <div
      className="emotion-0"
      data-element="bar"
      style={
        Object {
          "width": "40%",
        }
      }
    />
  </div>
  <div
    className="emotion-2"
  />
  <div
    className="emotion-1"
    data-element="step"
  >
    <div
      className="emotion-3"
      data-element="bar"
      style={
        Object {
          "width": "0",
        }
      }
    />
  </div>
  <div
    className="emotion-2"
  />
  <div
    className="emotion-1"
    data-element="step"
  >
    <div
      className="emotion-3"
      data-element="bar"
      style={
        Object {
          "width": "0",
        }
      }
    />
  </div>
  <div
    className="emotion-2"
  />
  <div
    className="emotion-1"
    data-element="step"
  >
    <div
      className="emotion-3"
      data-element="bar"
      style={
        Object {
          "width": "0",
        }
      }
    />
  </div>
</div>
`;

exports[`SteppedProgressBar matches snapshot when is small 1`] = `
@keyframes animation-0 {
  0% {
    background-position: -1200px 0;
  }

  100% {
    background-position: 1200px 0;
  }
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

.emotion-3 {
  position: absolute;
  z-index: 2;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background: #879ba6;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
}

.emotion-3::after {
  background: #879ba6;
  border-radius: 0 !important;
  background-image: linear-gradient( to right, #879ba6 8%, #d3dce6 18%, #879ba6 33% ) !important;
}

.emotion-2 {
  height: 100%;
  width: 1px;
  background: #fff;
}

.emotion-0 {
  position: absolute;
  z-index: 2;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background: #879ba6;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  overflow: hidden;
  border-radius: 0 !important;
  position: relative;
  pointer-events: none;
}

.emotion-0::after {
  background: #879ba6;
  border-radius: 0 !important;
  background-image: linear-gradient( to right, #879ba6 8%, #d3dce6 18%, #879ba6 33% ) !important;
}

.emotion-0::after {
  content: ' ';
  position: absolute;
  z-index: 9;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  background: #d3dce6;
  border-radius: 4px;
  overflow: hidden;
  background-image: linear-gradient( to right, #d3dce6 8%, #eaeff4 18%, #d3dce6 33% );
  background-size: 1200px 100%;
  -webkit-animation: animation-0 2s forwards infinite linear;
  animation: animation-0 2s forwards infinite linear;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 8px;
  width: 100%;
  position: relative;
  border-radius: 8px;
  overflow: hidden;
  background: #eaeff4;
  height: calc(8px / 2);
  border-radius: calc(8px / 2);
}

.emotion-11 [data-element='step']:first-child [data-element='bar'] {
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
}

.emotion-11 [data-element='step']:last-child [data-element='bar'] {
  border-top-right-radius: 8px;
  border-bottom-right-radius: 8px;
}

.emotion-11 [data-element='step']:first-child [data-element='bar'] {
  border-top-left-radius: calc(8px / 2);
  border-bottom-left-radius: calc(8px / 2);
}

.emotion-11 [data-element='step']:last-child [data-element='bar'] {
  border-top-right-radius: calc(8px / 2);
  border-bottom-right-radius: calc(8px / 2);
}

<div
  className="emotion-11"
>
  <div
    className="emotion-1"
    data-element="step"
  >
    <div
      className="emotion-0"
      data-element="bar"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="emotion-2"
  />
  <div
    className="emotion-1"
    data-element="step"
  >
    <div
      className="emotion-3"
      data-element="bar"
      style={
        Object {
          "width": "0",
        }
      }
    />
  </div>
  <div
    className="emotion-2"
  />
  <div
    className="emotion-1"
    data-element="step"
  >
    <div
      className="emotion-3"
      data-element="bar"
      style={
        Object {
          "width": "0",
        }
      }
    />
  </div>
  <div
    className="emotion-2"
  />
  <div
    className="emotion-1"
    data-element="step"
  >
    <div
      className="emotion-3"
      data-element="bar"
      style={
        Object {
          "width": "0",
        }
      }
    />
  </div>
</div>
`;
